src/StepAdaptationController/src/StepAdaptationController.cpp

```diff
@@ -0,0 +1,529 @@
+
+// std
+#define NOMINMAX
+#include <algorithm>
+
+// yarp
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Value.h>
+
+// iDynTree
+#include <iDynTree/Core/EigenSparseHelpers.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/SpatialAcc.h>
+#include <iDynTree/Core/Direction.h>
+#include <iDynTree/Core/CubicSpline.h>
+#include <iDynTree/Core/Twist.h>
+
+#include <WalkingControllers/StepAdaptationController/StepAdaptationController.hpp>
+#include <WalkingControllers/YarpUtilities/Helper.h>
+#include <WalkingControllers/iDynTreeUtilities/Helper.h>
+
+// eigen
+#include <Eigen/Sparse>
+
+typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> MatrixXd;
+using namespace WalkingControllers;
+
+typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> MatrixXd;
+using namespace WalkingControllers;
+
+StepAdaptationController::StepAdaptationController()
+    : m_inputSize(5)
+    , m_numberOfConstraints(7)
+    , m_xPositionsBuffer(2)
+    , m_yPositionsBuffer(2)
+    , m_zFirstPiecePositionsBuffer(3)
+    , m_zSecondPiecePositionsBuffer(2)
+    , m_yawsBuffer(2)
+    , m_timesBuffer(2)
+    , m_zFirstPieceTimesBuffer(3)
+    , m_zSecondPieceTimesBuffer(2)
+{
+
+    m_constraintsMatrix.resize(m_numberOfConstraints, m_inputSize);
+    m_upperBound.resize(m_numberOfConstraints);
+    m_lowerBound.resize(m_numberOfConstraints);
+    m_gradient.resize(m_inputSize);
+
+    // set the constant elements of the constraint matrix
+    m_constraintsMatrix(0, 0) = 1;
+    m_constraintsMatrix(1, 1) = 1;
+    m_constraintsMatrix(0, 3) = 1;
+    m_constraintsMatrix(1, 4) = 1;
+    m_constraintsMatrix(6, 2) = 1;
+
+    m_hessianMatrix.resize(m_inputSize, m_numberOfConstraints);
+    m_solution.resize(m_inputSize);
+
+    // qpoases
+    m_QPSolver_qpOASES = std::make_unique<qpOASES::SQProblem>(m_inputSize,
+                                                              m_numberOfConstraints);
+
+    m_QPSolver_qpOASES->setPrintLevel(qpOASES::PL_LOW);
+    m_isFirstTime = true;
+}
+
+bool StepAdaptationController::initialize(const yarp::os::Searchable &config)
+{
+
+    if(!YarpUtilities::getVectorFromSearchable(config, "next_zmp_position_weight",  m_zmpPositionWeight))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the vector";
+        return false;
+    }
+
+    if(!YarpUtilities::getVectorFromSearchable(config, "next_dcm_offset_weight",  m_dcmOffsetWeight))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the vector";
+        return false;
+    }
+
+    if(!YarpUtilities::getNumberFromSearchable(config, "sigma_weight", m_sigmaWeight))
+    {
+        yError() << "[StepAdaptationController::initialize] Unable to get the number";
+        return false;
+    }
+
+    m_feetExtendedPolygon.resize(2);
+    iDynTree::Polygon foot;
+    iDynTree::Vector4 nextZmpConstraintBoundLeftFoot  ;
```

Comment:
Can you remove the white space before ``;``?

Reply:
Done!